src/pages/_app.js

```diff
@@ -1,10 +1,27 @@
-import GlobalStyle from "../../styles";
+import { useState, useEffect } from 'react';
+import { useRouter } from "next/router";
+import transactionsData from '../data/transactions.json';
 
 export default function App({ Component, pageProps }) {
+  const [transactions, setTransactions] = useState(transactionsData);
+  const [successfulDeleted, setSuccessfulDeleted] = useState(false);
+  const router = useRouter();
+
+  const handleDelete = (id) => {
+    setTransactions((prevItems) => prevItems.filter(item => item.id !== id));
+    router.push('/')
+    setSuccessfulDeleted(true);
+  };
+
+  useEffect(() => {
+    if (successfulDeleted) {
+      setTimeout(() => {setSuccessfulDeleted(false)}, 5000)
+    }
+  },[successfulDeleted]);
+
   return (
     <>
-      <GlobalStyle />
-      <Component {...pageProps} />
+      <Component {...pageProps} handleDelete={handleDelete} transactions={transactions} successfulDeleted={successfulDeleted} />
```

Comment:
```suggestion
      <Component {...pageProps} onDelete={handleDelete} transactions={transactions} successfulDeleted={successfulDeleted} />
```

lifting up state naming convention